docs/2-Interviews/answering_technical_questions.md

```diff
@@ -1,39 +1,129 @@
+  - You don't have to get *coding* right away. In fact, you probably shouldn't
+  - Some things you can start with is to ask questions, draw some examples, describe possible approaches, write pseudocode
+- Ask questions!!!
+  - Clarify the problem to make sure you know what you're being asked (explain it back)
+  - Clarify requirements and assumptions (ask about edge cases, or clarifying details). E.g. shuffle a deck of cards - do I know how many cards? what does shuffling mean in this case?
+  - Go over 1-2 small test cases to make sure you know how the algorithm will work
+  - Note: figuring out what to ask is something you learn over time. Typical questions include: "what is the input size?", "is the input sorted?", "can I assume the input is always valid/well-formatted?", "how much memory do I have?".
+- Explain to the interviewer how you intend to approach and solve the problem at a conceptual level before you start writing code
+  - It helps you clarify your own thoughts
+  - It helps the interviewer follow along when you start coding
+  - It allows the interviewer to correct you if you misunderstood the question
+  - It allows the interviewer to jump in if the solution is totally off-track. Better to know early on before you write too much code going down the wrong path!
+  - Write stuff down as you think so they can see that you can organize your thoughts
+- It's okay to start with an inefficient approach, even brute force! Just get started, you can optimize things later - better to get a simple solution than no solution. Just don't forget to tell the interviewer that you're going for the brute force route. You don't want the interviewer to think that you think O(c^n) solution is efficient. So just tell the interviewer that you are aware that this is not the most efficient solution, but that writing even an inefficient approach will give you insight into the problem
+
+### Algorithms: solving the problem
+
+- Explain your thought process. Keep talking throughout the interview. Humans (interviewers included) tend to be uncomfortable with long silences for whatever reason.
+  - It's perfectly fine to take time to think. Just state it. The point is to avoid suddenly going silent. Useful phrases include "I'm going to take a minute to think about possible solutions" or "I'm worried that my solution does not handle this edge case. Let me think about this for a minute". Think of it as if you were narrating your own thought process: "I am wondering whether I should use approach A or approach B"
+  - Always explain what you are coding next. It doesn't have to be a long-winded explanation, often a sentence is enough. For example, "next, I'm going to find the object in this array with the largest index" before writing the 3-5 lines of code that achieves it
```

Comment:
This can be helpful if you are going in the wrong direction. If finding the object in the array with the largest index won't help you solve the problem, your interviewer might pipe in and encourage you to try something else before you waste time coding it up